### PR TITLE
feat: add handler for docker auth

### PR DESCRIPTION
**Pull request by rutujasurve94**
- [ ] fetch docker username and password from the env

- [ ] base 64 encode and return

### CODE REVIEW

File: lambda_handlers/tests/unit/test_docker_auth.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    return logger
+
+
+def test_docker_auth(monkeypatch):
+
+    monkeypatch.setenv('docker_username', 'test_username')
+    monkeypatch.setenv('docker_password', 'test_password')
+   
+    returned_body = lambda_handler(None, None)
+    assert returned_body['statusCode'] == 200
```
Comment (deepankarm) [inline lambda_handlers/tests/unit/test_docker_auth.py:22]: you can add an assert to check the `username` and `password` too

File: lambda_handlers/docker_auth.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+
+def lambda_handler(event, context):
+    logger = get_logger(context='docker_auth')
+
+    docker_username = os.environ['docker_username']
+    docker_password = os.environ['docker_password']
+
+    encoded_docker_username = base64.b64encode(docker_username.encode('utf-8'))
```
Comment (deepankarm) [inline lambda_handlers/docker_auth.py:30]: as discussed, we will be storing the env variable as `base64` encoded strings already. there's no need to encode it again

**Review by deepankarm — CHANGES_REQUESTED** [review]

**Comment by deepankarm** [comment]
Please make sure to test that it works on AWS Lambda. Also, we need to add this to the deployment script.

**Comment by rutujasurve94** [comment]
Yes, tested this on `AWS lambda` as well, receiving response:
```
Response:
{
  "isBase64Encoded": true,
  "headers": {
    "Content-Type": "application/json"
  },
  "statusCode": 200,
  "body": "{\"docker_username\": \"test_username\", \"docker_password\": \"test_password\"}"
}

Request ID:
"90048e97-d4c4-4ed2-b44f-f8a4adb6a95f"

Function logs:
START RequestId: 90048e97-d4c4-4ed2-b44f-f8a4adb6a95f Version: $LATEST
END RequestId: 90048e97-d4c4-4ed2-b44f-f8a4adb6a95f
REPORT RequestId: 90048e97-d4c4-4ed2-b44f-f8a4adb6a95f	Duration: 1.28 ms	Billed Duration: 2 ms	Memory Size: 128 MB	Max Memory Used: 50 MB	Init Duration: 123.93 ms	
```

File: .github/workflows/hub-api-cd.yml [hunk 3]
```diff
@@ -8,6 +8,8 @@ on:
       - closed
     paths:
       - 'lambda_handlers/*.py'
+    paths-ignore:
```
Comment (deepankarm) [inline .github/workflows/hub-api-cd.yml:11]: Both `paths` & `paths-ignore` cannot be in the same workflow - [link](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#excluding-paths). Anyway, `paths-ignore` is not needed, as we are only considering `*.py` in `paths`

File: hubapi/deployment.py [hunk 4]
```diff
@@ -23,7 +23,9 @@ def read_environment():
     password = str_to_ascii_to_base64_to_str(os.environ.get('JINA_DB_PASSWORD'))
     database_name = str_to_ascii_to_base64_to_str(os.environ.get('JINA_DB_NAME'))
     collection_name = str_to_ascii_to_base64_to_str(os.environ.get('JINA_DB_COLLECTION'))
-    return hostname, username, password, database_name, collection_name
+    docker_username = str_to_ascii_to_base64_to_str(os.environ.get('JINA_DOCKER_USERNAME'))
```
Comment (deepankarm) [inline hubapi/deployment.py:26]: Make sure these env variables are stored as secrets in Github.

File: .github/workflows/hub-api-cd.yml [hunk 5]
```diff
@@ -22,6 +24,7 @@ env:
   HUB_LIST_LAMBDA: hubapi_list
```
Comment (deepankarm) [inline .github/workflows/hub-api-cd.yml:24]: Need to add `JINA_DOCKER_USERNAME` & `JINA_DOCKER_PASSWORD` env variables here (to read from secrets)

File: hubapi/deployment.py [hunk 6]
```diff
@@ -44,7 +46,9 @@ def read_environment():
 @click.option('--deployment-stage',
               default='dev',
               help='Deployment stage for API Gateway (Default - dev)')
-def trigger(list_deployment_zip, push_deployment_zip, authorize_deployment_zip, key_id,
+@click.option('--docker-cred-deployment-zip',
```
Comment (deepankarm) [inline hubapi/deployment.py:49]: `click` reads env variables in the order decorator gets called & params are mentioned. Please correct this

**Review by deepankarm — CHANGES_REQUESTED** [review]

File: hubapi/deployment.py (hunk 4 above)
Comment (rutujasurve94) [reply hubapi/deployment.py:26]: Done

**Comment by deepankarm** [comment]
**Please test this again before merging, as it might break existing hubapi**

**Review by deepankarm — APPROVED** [review]
**Please test this again on AWS before merging, as it might break existing hubapi**

**Review by deepankarm — APPROVED** [review]

**Pull request merged**